File: Lab-Setup-Source-Code/Monitor.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    FILE *fptr;
    char filename[] = "process_log.txt";
    char c;
    time_t t;
    struct tm *tm_info;

    // Open file for appending
    fptr = fopen(filename, "a");

    if (fptr == NULL) {
        printf("Cannot open file \n");
        exit(0);
    }

    // Get current time
    time(&t);
    tm_info = localtime(&t);

    // Write timestamp to the file
    fprintf(fptr, "\nList of processes on the system\n");
    fprintf(fptr, "Timestamp: %s", asctime(tm_info));

    // Run system command to get all processes
    fprintf(fptr, "List of processes on the system:\n");
    system("tasklist");
    fprintf("\n");

    // Close file
    fclose(fptr);

    printf("Data appended to %s\n", filename);

    return 0;
}

File: Lab-Setup-Source-Code/Monitor.cs
using System;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        string filename = "process_log.txt";
        string tempfilename = "temp_process_log.txt";

        // Open file for appending
        StreamWriter fptr;
        try
        {
            fptr = File.AppendText(filename);
        }
        catch (Exception e)
        {
            Console.WriteLine("Cannot open file: " + e.Message);
            return;
        }

        // Get current time
        DateTime now = DateTime.Now;

        // Write timestamp to the file
        fptr.WriteLine("\nList of processes on the system");
        fptr.WriteLine("Timestamp: " + now.ToString());

        // Run system command to get all processes and save in a temporary file
        System.Diagnostics.Process.Start("CMD.exe", "/C tasklist > temp_process_log.txt").WaitForExit();

        // Append contents of temporary file to the log file
        try
        {
            using (StreamReader tempfptr = new StreamReader(tempfilename))
            {
                string line;
                while ((line = tempfptr.ReadLine()) != null)
                {
                    fptr.WriteLine(line);
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine("Cannot open temporary file: " + e.Message);
            return;
        }

        File.Delete(tempfilename);

        // Close file
        fptr.Close();

        Console.WriteLine("Data appended to " + filename);
    }
}